pkg/distributor/distributor.go

```diff
@@ -218,6 +238,44 @@ func shardByAllLabels(userID string, labels []client.LabelPair) (uint32, error)
 	return h.Sum32(), nil
 }
 
+// Remove the replica label from a slice of LabelPairs if it exists.
+func removeReplicaLabel(replica []byte, labels *[]client.LabelPair) {
+	for i := 0; i < len(*labels); i++ {
+		pair := (*labels)[i]
+		if pair.Name.Equal(replica) {
+			*labels = append((*labels)[:i], (*labels)[i+1:]...)
+			return
+		}
+	}
+}
+
+// Returns an error if we should not accept samples based on the HA labels found in ts. Boolean value
+// is whether we found both HA labels. We use the boolean value to know if we want to remove the
+// replica label.
+func (d *Distributor) checkSample(ctx context.Context, ts client.PreallocTimeseries) (bool, error) {
```

Comment:
This error returned is never used, and is only one of two values - nil or `fmt.Errorf...`.  Probably want to use a bool.

Reply:
Do we want to look for a way to have `Push` return a 202 when we don't want to accept the sample? I've just had it return `return &client.WriteResponse{}, nil` like we're accepting the sample which AFAICT results in a 200.

I didn't see a way to do this with the http grpc stuff you mentioned.